Rubberduck.Inspections/Abstract/QuickFixBase.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NLog;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Parsing.VBA;
+
+namespace Rubberduck.Inspections.Abstract
+{
+    public abstract class QuickFixBase : IQuickFix
+    {
+        private readonly ILogger _logger = LogManager.GetCurrentClassLogger();
+        private HashSet<Type> _supportedInspections;
+        public IReadOnlyCollection<Type> SupportedInspections => _supportedInspections.ToList();
+
+        protected QuickFixBase(params Type[] inspections)
+        {
+            RegisterInspections(inspections);
+        }
+
+        public void RegisterInspections(params Type[] inspections)
+        {
+            if (!inspections.All(s => s.GetInterfaces().Any(a => a == typeof(IInspection))))
+            {
+#if DEBUG
+                throw new ArgumentException($"Parameters must implement {nameof(IInspection)}", nameof(inspections));
+#else
+                inspections.Where(s => s.GetInterfaces().All(i => i != typeof(IInspection))).ToList()
+                    .ForEach(i => _logger.Error($"Type {i.Name} does not implement {nameof(IInspection)}"));
+#endif
+            }
+
+            _supportedInspections = inspections.ToHashSet();
```

Comment:
Now that you're logging as per @Vogel612's suggestion, I wonder if the `RegisterInspection` should at least communicate whether it encountered an error so the calling code won't have incorrect assumptions about whether it was successful in registering the inspections?

Reply:
@bclothier the calling code doesn't make any assumptions about the success of that operation

Reply:
I might have lost the plot but if the point is enabling users to load their custom inspections dynamically, among with those that ships with RD, doesn't it imply a need to message to the users there was a problem in registering the custom inspections? The message can refer to the log but the user has to at least know, no?

Reply:
@bclothier the user will know as soon as they attempt to execute their custom inspections and quickfixes. Anything else would be premature and only relevant to a minuscle fraction of the userbase.

Who's going to develop their own inspections and quickfixes for rubberduck? And if there's only few of them, why should we "burden" the bulk of our users with diagnostic information for this "rare" case?

Reply:
If they test it in Debug mode, as they should, they get an exception.